climada/engine/impact.py

```diff
@@ -510,6 +514,79 @@ def write_sparse_csr(self, file_name):
         np.savez(file_name, data=self.imp_mat.data, indices=self.imp_mat.indices,
                  indptr=self.imp_mat.indptr, shape=self.imp_mat.shape)
 
+    def write_netcdf(self, path=None, attrs=None, with_imp_mat=True, **kwargs):
+        """Write this instance into a (rasterized) netcdf via a DataArray.
+
+        Parameters:
+            path (str, optional): Where to write the resulting netCDF to. If
+                None, the method stands in as a to_dataset
+            attrs (str, optional): Set as an attribute; Dataset.to_netcdf does
+                not handle dimensionless variables properly, so GDAL cannot
+                handle contained projection information. Still worthwhile for
+                human reference though.
+            with_imp_mat (bool): If true, read impact values from imp_mat
+                instead of eai_exp
+        Returns:
+            xarray.DataArray
+        """
+
+        lats, lons = unique_lat_lon_from_meta(self.exposures.meta)
+        coords = {'latitude': lats, 'longitude': lons}
+
+        if with_imp_mat:
+            data = np.ndarray((self.event_id.size, lats.size, lons.size))
+            for i, event_id in enumerate(self.event_id):
+                data[i] = self.rasterize(event_id, **kwargs)
+            coords['event_id'] = self.event_id
+            dims = ['event_id', 'latitude', 'longitude']
+        else:
+            data = self.rasterize(meta = self.exposures.meta, **kwargs)
+            dims = ['latitude', 'longitude']
+
+        if attrs is None:
+            attrs = {
+                'crs': 'EPSG:' + str(self.exposures.meta['crs'].to_epsg())
+            }
+
+        imp_ds = xr.DataArray(data, coords, dims, 'impact', attrs)
+
+        if path:
+            imp_ds.to_netcdf(path)
+
+        return imp_ds
+
+    def rasterize(self, event_id=None, meta=None, drop_zero=True):
+        """Rasterize this impact instance, returning a numpy ndarray with no
+        associated metadata. See to_dataarray for usage example.
```

Comment:
Couldn't this method be private?
What is `to_dataarray` referring to?
In any case, if documentation is "outsourced" the outer source must be linked to in the generated docs.

Reply:
I did make it private at first, but thought that users might profit from the rasterization util. Will change that.

`to_dataarray` stems from a time where I did (as you had suggested) have a method that simply converted the instance to a dataarray. Sorry for my oversight.

How do you link to other docstrings? Is that in rst syntax?